rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt

```diff
@@ -159,17 +159,29 @@ private fun valueToString(value: Value, type: Type): String {
         is CharacterType -> return s
 
         is NumberType -> {
-            if (value is IntValue) {
-                val zeros = "0".repeat(type.numberOfDigits - s.length)
-                zeros + s
-            } else {
-                s = s.replace(".", "")
-                val zeros = "0".repeat(type.numberOfDigits - s.length)
-                zeros + s
+            when (value) {
```

Comment:
Too much duplicated code. `IntValue` and `DecimalValue` extend `NumberValue` that provides a method `isNegative.` Follow my advice and see if you are able to make code cleaner.

Reply:
Fixed with isNegative() method